Dockerfile

```diff
@@ -1,27 +1,29 @@
-FROM eclipse-temurin:17
+FROM eclipse-temurin:17-jre-alpine
 
 ENV TERRAFORM_VERSION=1.4.4
 ENV XPANSE_USER=xpanse
 ENV XPANSE_HOME_DIR=/home/xpanse
 
 # Install dependencies
-RUN apt-get update && \
-    apt-get install unzip wget -y
+RUN apk update && \
+    apk add --no-cache unzip wget openssh-keygen openssl
 
 # Install Terraform
 RUN wget https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
 RUN unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip
 RUN mv terraform /usr/bin/terraform
 
 # Create xpanse user
-RUN groupadd -r -g 2000 ${XPANSE_USER} && useradd -m -d ${XPANSE_HOME_DIR} -s /bin/bash -u 2000 -r -g ${XPANSE_USER} ${XPANSE_USER}
+RUN addgroup -S -g 2000 ${XPANSE_USER} && \
+    adduser -S -h ${XPANSE_HOME_DIR} -s /bin/sh -u 2000 -G ${XPANSE_USER} ${XPANSE_USER}
 USER ${XPANSE_USER}
 WORKDIR ${XPANSE_HOME_DIR}
 
 # Copy jar file
 COPY --chown=${XPANSE_USER}:${XPANSE_USER} runtime/target/xpanse-runtime-*.jar xpanse-runtime.jar
 
-# Generate random key
-RUN tr -dc 'a-zA-Z0-9' < /dev/urandom | head -c 32 > aes_sec
+# Generate RSA private key
+RUN openssl genpkey -algorithm RSA -out /home/xpanse/private_key.pem
```

Comment:
@jinyangyang222 this will still create the key in the image. 
We need something that will create the key only when the container i started.